language.go

```diff
@@ -65,35 +65,33 @@ func (language *GoLanguageModel) Prepare(model *surface.Model) {
 		m.HandlerName = "Handle" + m.Name
 		m.ProcessorName = m.Name
 		m.ClientName = m.Name
+		m.ResponsesTypeName = strings.Title(filteredTypeName(m.ResponsesTypeName))
 	}
+	gnostic_grpc.AdjustSurfaceModel(model, inputDocumentType)
 }
 
-func goParameterName(name string) string {
+func goParameterName(originalName string, t string) string {
+	name := gnostic_grpc.CleanName(originalName)
+	if len(name) == 0 {
+		name = gnostic_grpc.CleanName(t)
+	}
 	// lowercase first letter
 	a := []rune(name)
 	a[0] = unicode.ToLower(a[0])
 	name = string(a)
-	// replace dots with underscores
-	name = strings.Replace(name, ".", "_", -1)
-	// replaces dashes with underscores
-	name = strings.Replace(name, "-", "_", -1)
 	// avoid reserved words
 	if name == "type" {
 		return "myType"
 	}
 	return name
 }
 
-func goFieldName(name string) string {
-	name = strings.Replace(name, ".", "_", -1)
-	name = strings.Replace(name, "-", "_", -1)
-	name = snakeCaseToCamelCaseWithCapitalizedFirstLetter(name)
-	// avoid integers
-	if name == "200" {
-		return "OK"
-	} else if unicode.IsDigit(rune(name[0])) {
-		return "Code" + name
+func goFieldName(name string, t string) string {
+	name = gnostic_grpc.CleanName(name)
+	if len(name) == 0 {
+		name = gnostic_grpc.CleanName(t)
 	}
+	name = snakeCaseToCamelCaseWithCapitalizedFirstLetter(name)
```

Comment:
Inline this with the `return` statement please.

`return snakeCaseToCamelCaseWithCapitalizedFirstLetter(name)`

Side note: I know this method wasn't added in this PR, and no need to change it here, but that name is way too long. `snakeToCamelExported` is probably enough in my opinion (assuming the "CapitalizedFirstLetter" part is referring to the exported version of a field name).

Reply:
Done